.github/workflows/main-drivers.yml

```diff
@@ -0,0 +1,59 @@
+name: Collector Drivers CI
+
+on:
```

Comment:
Think we want to add `pull_request:` here, otherwise we won't be able to test the workflow if/when it breaks. AFAICT, the safeguards for running this on PRs are still there, so it should be OK for us to add it in.

Reply:
Agreed, but I think we only want to run it on PRs that request driver builds, rather than running it in general

Reply:
Yeah, that should be handled by the fact that we only build drivers that are missing from the cache bucket, right? Same way we handle it currently, or did you have something else in mind?

Reply:
What is exactly understood here under safeguards? If it's only conditions inside the pipeline to not do anything, unless `no-cache` labels is set, in long term this sounds a bit fragile to me. How about keeping it on master push only, and start the development workflow with "turn it on for PRs"?